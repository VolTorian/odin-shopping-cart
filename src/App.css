#root {
    text-align: center;
}

body {
    display: block;
}

.navbar {
    display: flex;
    background-color: cyan;
    justify-content: center;
    width: 100vw;
}

.navbar button {
    background-color: transparent;
}

.cards-section {
    display: flex;
    flex-wrap: wrap;
    gap: 50px;
    justify-content: center;
}

.item-card {
    width: 250px;
    height: 400px;
    border: solid black 1px;
    padding: 20px;
    display: flex;
    flex-direction: column;
}

.item-title {
    margin: .5rem 0;
    overflow: hidden;
    text-overflow: ellipsis;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
}

.item-card > .image-container {
    width: 100%;
    height: 200px;
}

.image-container > img {
    max-width: 100%;
    max-height: 100%;
    width: auto;
    height: auto;
}

.item-rating {
    margin-top: auto;
}

.item-rating, .item-price {
    margin-bottom: .5rem;
}

.item-card-buttons {
    display: flex;
    justify-content: center;
}

.item-card-buttons > button {
    padding: .5rem;
}

.item-card-buttons input {
    width: 2rem;
}

.cart-items-section {
    width: 80%;
    margin: auto;
}

.cart-item-listing {
    display: flex;
    padding: 20px;
    margin: auto;
    height: 100px;
}

.image-container {
    align-content: center;
}

.cart-item-listing > .image-container {
    width: 100px;
}

.cart-item-info {
    width: 100%;
}

.cart-item-info > h4 {
    margin: 0 1rem 1rem 1rem;
}

.item-details-modal {
    width: 80vw;
}