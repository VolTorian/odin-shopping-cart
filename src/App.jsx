import './App.css'

import { useEffect, useState } from 'react'
import { Outlet } from 'react-router';

import NavBar from "./NavBar";

function App() {
    const SHOP_ITEM_COUNT = 20;
    const [shopItems, setShopItems] = useState([]);

    useEffect(() => {
        fetchItems(SHOP_ITEM_COUNT);
    }, []);

    async function fetchItems(numberOfItems) {
        const initialList = new Array(numberOfItems).fill(null);
        setShopItems(initialList);

        for (let i = 1; i <= numberOfItems; i++) {
            fetch(`https://fakestoreapi.com/products/${i}`)
                .then((response) => response.json())
                .then((data) => {
                    setShopItems((prev) => {
                        const updated = [...prev];
                        updated[i - 1] = data;
                        // console.log(data.id)
                        return updated;
                    });
                });
        };
    }

    return (
        <>
            <NavBar />
            <Outlet context={[shopItems]} />
        </>
    );
}

export default App;