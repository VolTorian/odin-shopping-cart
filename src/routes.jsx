import App from "./App";
import Cart from "./Cart";
import Home from "./Home";
import Shop from "./Shop";

const routes = [
    {
        path: "/",
        element: <App />,
        children: [
            { path: "", element: <Home /> },
            { path: "shop", element: <Shop /> },
            { path: "cart", element: <Cart /> },
        ]
    }
];

export default routes;