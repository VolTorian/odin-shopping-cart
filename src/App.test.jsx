import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter, RouterProvider, createMemoryRouter } from 'react-router-dom';

import App from "./App";

import routes from "./routes";
const router = createMemoryRouter(routes);

describe("App component", () => {
    // it("renders navbar in app", () => {
    //     const { container } = render(
    //         <MemoryRouter>
    //             <App />
    //         </MemoryRouter>
    //     );

    //     expect(container).toMatchSnapshot();
    // });

    it("home button sends to home page", async () => {
        const user = userEvent.setup();

        render(
            <RouterProvider router={router} />
        );

        const homeButton = screen.getByRole("button", { name: "Home"});

        await user.click(homeButton);
        expect(screen.getByRole("heading").textContent).toMatch(/Home/i);
    });

    // it("shop button sends to shop page", async () => {
    //     const user = userEvent.setup();

    //     render(
    //         <MemoryRouter>
    //             <App />
    //         </MemoryRouter>
    //     );

    //     const shopButton = screen.getByRole("button", { name: "Shop"});

    //     await user.click(shopButton);
    //     expect(screen.getByRole("heading").textContent).toMatch(/Shop/i);
    // });

    // it("cart button sends to cart page", async () => {
    //     const user = userEvent.setup();

    //     render(<MemoryRouter>
    //         <App />
    //     </MemoryRouter>
    //     );

    //     const cartButton = screen.getByRole("button", { name: "Cart"});

    //     await user.click(cartButton);
    //     expect(screen.getByRole("heading").textContent).toMatch(/Cart/i);
    // });
});