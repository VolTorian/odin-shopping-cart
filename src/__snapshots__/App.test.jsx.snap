// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders navbar in app 1`] = `
<div>
  <div
    class="navbar"
  >
    <a
      data-discover="true"
      href="/"
    >
      <button>
        Home
      </button>
    </a>
    <a
      data-discover="true"
      href="/shop"
    >
      <button>
        Shop
      </button>
    </a>
    <a
      data-discover="true"
      href="/cart"
    >
      <button>
        Cart
      </button>
    </a>
  </div>
  <h3>
    Home
  </h3>
</div>
`;
